core/include/scipp/core/time_point.h

```diff
@@ -4,36 +4,74 @@
 #pragma once
 
 #include <stdint.h>
+#include <string>
 
 namespace scipp::core {
 
 /// Time-point similar to std::chrono::time point but without compile-time unit.
+/// The unit is determined by the Variable the time_point is stored in.
 class time_point {
 public:
-  constexpr time_point() noexcept = default;
-  constexpr time_point(const int64_t &d) noexcept : m_duration(d){};
-  constexpr time_point(const int32_t &d) noexcept
-      : m_duration(static_cast<int64_t>(d)){};
+  time_point() = default;
+  explicit time_point(int64_t duration) : m_duration{duration} {}
+  time_point(const time_point &other) = default;
+  time_point(time_point &&other) noexcept = default;
+  time_point &operator=(const time_point &other) = default;
+  time_point &operator=(time_point &&other) noexcept = default;
+  ~time_point() noexcept = default;
 
-  int64_t time_since_epoch() const noexcept { return m_duration; };
+  [[nodiscard]] int64_t time_since_epoch() const noexcept {
+    return m_duration;
+  };
+
+  friend time_point operator+(const time_point a, const int64_t b) {
```

Comment:
Why are all these friends instead of free functions?

Reply:
To hide them from overload resolution where not appropriate.

Reply:
Can you elaborate what the problem is?

Reply:
See the error message in https://godbolt.org/z/77e847

The overload off `A` is detected but not the one for `B`. This makes the errors more readable and maybe speeds up compilation.

Reply:
Hmm, interesting I did not know this! Do you recommend a reference for reading up on this technique?

Reply:
I searched for one but couldn't find any. I seem to remember this was in an episode of C++ Weekly. But I'm not sure anymore.